src/com/goide/highlighting/exitpoint/GoBreakStatementExitPointHandler.java

```diff
@@ -0,0 +1,149 @@
+  @NotNull private final PsiElement myTarget;
+  @Nullable private final GoBreakStatement myBreakStatement;
+  @Nullable private PsiElement myOwner;
+
+  private GoBreakStatementExitPointHandler(Editor editor,
+                                           PsiFile file,
+                                           @NotNull PsiElement target,
+                                           @Nullable GoBreakStatement breakStatement,
+                                           @Nullable PsiElement owner) {
+    super(editor, file);
+    myTarget = target;
+    myBreakStatement = breakStatement;
+    myOwner = owner;
+  }
+
+  @NotNull
+  @Override
+  public List<PsiElement> getTargets() {
+    return ContainerUtil.newSmartList(myTarget);
+  }
+
+  @Override
+  protected void selectTargets(List<PsiElement> targets, @NotNull Consumer<List<PsiElement>> selectionConsumer) {
+    selectionConsumer.consume(targets);
+  }
+
+  @Override
+  public void computeUsages(List<PsiElement> targets) {
+    final PsiElement breakStmtOwner;
+    if (myOwner != null) {
+      breakStmtOwner = myOwner;
+    }
+    else if (myBreakStatement != null) {
+      breakStmtOwner = getBreakStatementOwner(myBreakStatement);
+    }
+    else {
+      breakStmtOwner = null;
+    }
+    GoRecursiveVisitor visitor = new GoRecursiveVisitor() {
+      @Override
+      public void visitLabelDefinition(@NotNull GoLabelDefinition o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o);
+        }
+        super.visitLabelDefinition(o);
+      }
+
+      @Override
+      public void visitBreakStatement(@NotNull GoBreakStatement o) {
+        if (o == myBreakStatement || getBreakStatementOwner(o) == breakStmtOwner) {
+          addOccurrence(o);
+        }
+        super.visitBreakStatement(o);
+      }
+
+      @Override
+      public void visitSwitchStatement(@NotNull GoSwitchStatement o) {
+        if (o == breakStmtOwner) {
+          GoSwitchStart switchStart = o.getSwitchStart();
+          if (switchStart != null) {
+            addOccurrence(switchStart.getSwitch());
+          }
+        }
+        super.visitSwitchStatement(o);
+      }
+
+      @Override
+      public void visitForStatement(@NotNull GoForStatement o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o.getFor());
+        }
+        super.visitForStatement(o);
+      }
+
+      @Override
+      public void visitSelectStatement(@NotNull GoSelectStatement o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o.getSelect());
+        }
+        super.visitSelectStatement(o);
+      }
+    };
+    if (breakStmtOwner != null) {
+      PsiElement parent = breakStmtOwner.getParent();
+      if (parent instanceof GoCompositeElement) {
+        visitor.visitCompositeElement(((GoCompositeElement)parent));
+      }
+    }
+  }
+
+
+  @Nullable
+  private static PsiElement getBreakStatementOwner(@NotNull GoBreakStatement breakStmt) {
```

Comment:
You could also add an option to turn highlighting off.
In Scala plugin it's called type-aware highlighting